nonlineargot.py

```diff
@@ -0,0 +1,42 @@
+#!/bin/python3
+# -*- coding: utf-8 -*-
+
+import math
+import os
+
+#maximum number of characters in the string according to hackerrank
+NO_OF_CHARS = 100000
+
+# Complete the gameOfThrones function below.
+def gameOfThrones(s):
+    #create countermatrix full of zeros
+    count = [0]*NO_OF_CHARS
+
+    #count character occurences
+    for i in range(len(s)):
+        count[ord(s[i])] = count[ord(s[i])] + 1
```

Comment:
Yes, indeed, taking the unicode  of the character and using it as an index could work. However, the indexes returned from this functions are just some big integers. In the case of the english alphabet the codes are:

```python
ord('a') =  97 
ord('b') =  98
...
ord('z') =  122
```
The matrix `count` has a size of `NO_OF_CHARS`, but the maximum size of the input string is irrelevant to the size of this array. If the problem specified that the maximum size of the input string would be 100 characters, your code would throw an `out of index` error.

Reply:
And this is why you were getting the "Runtime Error" in Hackerrank